### PR TITLE
use git clone --depth 1 in CI

### PR DESCRIPTION
**Pull request by andres-erbsen**
Downloading the CI developments locally created more than a gigabyte of `.git` and took tens of minutes due to Github CPU throttling; this is an attempt at reducing both.

### CODE REVIEW

File: dev/ci/ci-common.sh [hunk 1]
```diff
@@ -111,11 +111,11 @@ git_download()
   if [ -d "$dest" ]; then
     echo "Warning: download and unpacking of $project skipped because $dest already exists."
   elif [[ $ov_url ]] || [ "$WITH_SUBMODULES" = "1" ] || [ "$CI" = "" ]; then
-    git clone "$giturl" "$dest"
+    git clone --depth 1 --branch "$ref" "$giturl" "$dest"
```
Comment (SkySkimmer) [inline dev/ci/ci-common.sh:114]: ref is not always a branch, on Coq's release branch it gets set to a commit

**Comment by Zimmi48** [comment]
Note that on CI, we do not use `git clone` for projects which do not use submodules for their dependencies (but rather download the GitHub tarball). It was decided that developers preferred to have a `git clone` locally, but it should be possible (I don't recall precisely) to override this setting with an environment variable... Anyway, if `--depth 1` doesn't prevent from creating and submitting an overlay, this change should not be a problem.

File: dev/ci/ci-common.sh (hunk 1 above)
Comment (andres-erbsen) [reply dev/ci/ci-common.sh:114]: Ah. The man page says `--branch` can also take tags, but I haven't found a reliable way to request a specific commit. So if we want this change at all, it'll need to fall back to clone-and-switch.

**Comment by andres-erbsen** [comment]
It is possible to commit on top of a shallow checkout and push to repositories that have the parent commit already. Pushing to a stale personal-Github copy of the repo of the upstream development would not work, and I don't know whether there is a way to bring that copy up to date without downloading the history and reuploading it. It's still possible to `git fetch --unshallow` after committing the change and seeing the push fail due to lack of shared history.

**Comment by SkySkimmer** [comment]
Github has an update button in the web interface these days so that's an alternative to fetch unshallow:
Screenshot_20221207_154632

File: dev/ci/ci-common.sh (hunk 1 above)
Comment (andres-erbsen) [reply dev/ci/ci-common.sh]: Actually, `init` and `pull` seem to do the trick for the repositories currently in CI.

**Comment by ejgallego** [comment]
> Anyway, if --depth 1 doesn't prevent from creating and submitting an overlay, this change should not be a problem.

Indeed the intended git checkout for developers is full, as it is intended to be a full developer tree on top of which overlays are pushed. Moreover, in the future we should be able to switch branches for the overlays too.

**Pull request closed without merging**